cmd/syncthing/support_bundle.go

```diff
@@ -0,0 +1,143 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"archive/zip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/syncthing/syncthing/lib/config"
+)
+
+// AddBufferFile adds new json to fileByte(name, bytes.Buffer)
+func addBufferFile(arrayFile []fileByte, name string, json []byte) []fileByte {
```

Comment:
Instead of using this function, it's more idiomatic Go to just

```
arrayFile = append(arrayFile, fileEntry{name: name, data: data})
```

where `fileEntry` is something like

```
type fileEntry struct {
  name string
  data []byte
}
```

I don't think keeping a buffer directly in the `fileByte` (as it's called now) here adds much value. Instead, the buffer handling can be done as necessary by the creator / user of these structs.